src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -131,30 +138,71 @@ private void processRequests(HttpServerRequest httpServerRequest) {
 
         switch (requestType){
             case PRODUCE:
-                ConnectionEndpoint endpoint = this.endpoints.get(httpServerRequest.connection());
-
-                SourceBridgeEndpoint source = endpoint.getSource();
+                SourceBridgeEndpoint source = this.httpSourceEndpoints.get(httpServerRequest.connection());
 
                 if (source == null) {
                     source = new HttpSourceBridgeEndpoint(this.vertx, this.bridgeConfigProperties);
                     source.closeHandler(s -> {
-                        endpoint.setSource(null);
+                        this.httpSourceEndpoints.remove(httpServerRequest.connection());
                     });
                     source.open();
-                    endpoint.setSource(source);
+                    this.httpSourceEndpoints.put(httpServerRequest.connection(), source);
                 }
                 source.handle(new HttpEndpoint(httpServerRequest));
 
                 break;
 
+            //create a sink endpoint and initialize consumer
+            case CREATE:
+                SinkBridgeEndpoint<?,?> sink = new HttpSinkBridgeEndpoint<>(this.vertx, this.bridgeConfigProperties);
+                sink.closeHandler(s -> {
+                    this.httpSinkEndpoints.remove(sink);
+                });
+
+                sink.open();
+
+                sink.handle(new HttpEndpoint(httpServerRequest), consumerId -> {
+                    this.httpSinkEndpoints.put(consumerId.toString(), sink);
+                });
+
+                break;
+
+            case SUBSCRIBE:
+            case CONSUME:
+            case OFFSETS:
+                String instanceId = PathParamsExtractor.getConsumerSubscriptionParams(httpServerRequest).get("instance-id");
+
+                final SinkBridgeEndpoint sinkEndpoint = this.httpSinkEndpoints.get(instanceId);
+
+                if (sinkEndpoint != null) {
+                    sinkEndpoint.handle(new HttpEndpoint(httpServerRequest));
+                } else {
+                    throw new RuntimeException("no conusmer instance found with this id");
+                }
+                break;
+
+            case DELETE:
+                String deleteInstanceID = PathParamsExtractor.getConsumerDeletionParams(httpServerRequest).get("instance-id");
+
+                final SinkBridgeEndpoint deleteSinkEndpoint = this.httpSinkEndpoints.get(deleteInstanceID);
+
+                if (deleteSinkEndpoint != null) {
+
+                    deleteSinkEndpoint.handle(new HttpEndpoint(httpServerRequest));
+
+                    this.httpSinkEndpoints.remove(deleteInstanceID);
+                } else {
+                    throw new NullPointerException("no conusmer instance found with this id");
```

Comment:
you should not throw an exception but replying to the HTTP client with a "404 Not Found" status code even because we want to avoid an NPE but you are raising an NPE :-)